compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -48,7 +48,7 @@ object ProtoTypes {
 
     private def disregardProto(pt: Type)(implicit ctx: Context): Boolean = pt.dealias match {
       case _: OrType => true
-      case pt => pt.isRef(defn.UnitClass)
+      case ptd => ptd.isRef(defn.UnitClass)
```

Comment:
the 'case' pt shadows the outer pt, which may not be an issue here, but could crop up later if the function became more complex, I suppose. Happy to take it out though.